### PR TITLE
Restore sun halo

### LINKED ISSUE #541: Sun halo in different starfield rendering modes
Both in 1.6 and 1.7 versions. Depending on starfield's style a sun halo either present or not:
No halo with scaled disks:
disks
Halo present with points and fuzzy points:
fuzzy
points

1.7 had the same behaviour before #533. Now it has less bright halo in «points» and «fuzzy points» modes.

Quite interesting is that Alexell's version didn't have this bug. UPD: when vp&fp programs doesn't present, so older strarfied with quads is in use.

### PR DESCRIPTION
**Pull request by levinli303**
Just copied some of the code back. Fix #541  ?

### CODE REVIEW

**Comment by 375gnu** [comment]
old method used quads instead of points so because of that additional math was required. and here we still have the old bug with halo disappearing when a sun goes away from the screen

**Comment by levinli303** [comment]
yeah, maybe we should take the halo size into account when we decide if a star should be visible too. and by the way, we might need to draw quad when the point size is too large. for example, some ANGLE implementation limits point size to 64.

**Comment by 375gnu** [comment]
i tried to restore the original code but without success

**Comment by SevenSpheres** [comment]
Tested now but I see this has been marked as draft now - anyway for me this only shows a bright point at a star's center, it doesn't restore the "halo" on the whole star. (Screenshot has FarGetaNik's gstar texture and not the default)

sol849

**Comment by levinli303** [comment]
@375gnu  Updated to draw quads when point size > 64, or should we always draw quad in this method? This does not fix the issue where the halo disappearing when a sun goes away from the screen, but maybe we can still have this fix in?

Simulator Screen Shot - iPhone 6 - 2020-10-24 at 18 03 50

**Comment by levinli303** [comment]
@SevenSpheres can you try this https://ci.appveyor.com/project/375gnu/celestia/builds/35932620/artifacts and see if it fixes your problem. you also need to download new shaders (largestar*) from https://github.com/levinli303/Celestia/tree/restore-sun-halo/shaders

**Comment by SevenSpheres** [comment]
> @SevenSpheres can you try this https://ci.appveyor.com/project/375gnu/celestia/builds/35932620/artifacts and see if it fixes your problem. you also need to download new shaders (largestar*) from https://github.com/levinli303/Celestia/tree/restore-sun-halo/shaders

Yes, that works!

File: src/celengine/render.cpp [hunk 1]
```diff
@@ -140,6 +140,8 @@ static const float MaxAsterismLinesConstDist   = 600.0f;
 static const float MaxAsterismLabelsDist = 20.0f;
 static const float MaxAsterismLinesDist  = 6.52e4f;
 
+static const float MaxPointSize  = 64.0f;
```
Comment (375gnu) [inline src/celengine/render.cpp:143]: don't hardcode this. use
```
GLint pointSizeRange[2];
glGetIntegerv(GL_SMOOTH_POINT_SIZE_RANGE, pointSizeRange);
MaxPointSize = pointSizeRange[1];
```
somewhere in `Renderer::Renderer()`

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celengine/render.cpp (hunk 1 above)
Comment (levinli303) [reply src/celengine/render.cpp:143]: maybe we can put it in gl::init?

Comment (levinli303) [reply src/celengine/render.cpp:143]: and should we use GL_ALIASED_POINT_SIZE_RANGE for gles?

Comment (375gnu) [reply src/celengine/render.cpp:143]: > maybe we can put it in gl::init?

sounds like a good idea



> and should we use GL_ALIASED_POINT_SIZE_RANGE for gles?

yes

Comment (levinli303) [reply src/celengine/render.cpp]: done

**Comment by levinli303** [comment]
https://ci.appveyor.com/project/375gnu/celestia/builds/35948238/artifacts @SevenSpheres  maybe you can test again?

I tested on macOS and iOS it works well. Also works on macCatalyst with ANGLE (max point size is 64)

**Review by SevenSpheres — APPROVED** [review]

**Comment by LukeCEL** [comment]
Is it possible to apply the sun halo effect in "Scaled Disk" mode?

**Review by 375gnu — APPROVED** [review]

**Pull request merged**